### PR TITLE
Auto Rebase Github Action

### PR DESCRIPTION
**Pull request by awasum**
#### Is this adding or improving a _feature_ or fixing a _bug_?

Adding a new feature: To automatically Rebase PRs during review.

#### What's the new behavior?

To automatically Rebase PRs during review, write /rebase as comment on a PR. 

#### How does this change work?

It reduces time to manually do the rebase and push upstream. It also updates the latest changes from master unto your PR branch

#### Have you checked that...?

- [ ] The new code matches the existing patterns and styles.
- [ ] The tests pass with `yarn test`.
- [ ] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [ ] The relevant examples still work. (Run examples with `yarn start`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: https://github.com/ianstormtaylor/slate/issues/3764
Reviewers: @ianstormtaylor

### CODE REVIEW

**Comment by timbuckley** [comment]
Looks nice, but the `Allow Edits and access to secrets by Maintainers` option might only be done via @ianstormtaylor, and I'm not sure he would approve. CC: @CameronAckermanSEL

**Comment by awasum** [comment]
> Looks like a nice addition, but the required `Allow Edits and access to secrets by Maintainers` option might only be done via @ianstormtaylor, and I'm not sure he would approve. CC: @CameronAckermanSEL

This needs to be done by the author sending a PR in not the repo owner receiving the PR. This is so that github action can auto rebase on the PR author's behalf. I don't know if you understand me?

**Comment by awasum** [comment]
@timbuckley . I corrected the description for the PR. I hope this is clearer than before. CC @ianstormtaylor

File: .github/workflows/rebase.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    if: github.event.issue.pull_request != '' && startsWith(github.event.comment.body, '/rebase')
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+      with:
+        fetch-depth: 0
+    - name: Automatic Rebase
+      uses: cirrus-actions/rebase@1.3.1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (timbuckley) [inline .github/workflows/rebase.yml:18]: Personally, I prefer `env` to be extracted to the top of the workflow, so it's immediately clear what env vars are needed for the job.

```suggestion
name: Automatic Rebase

on:
  issue_comment:
    types: [created]

env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

jobs:
  rebase:
    name: Rebase
    if: github.event.issue.pull_request != '' && startsWith(github.event.comment.body, '/rebase')
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
      with:
        fetch-depth: 0
    - name: Automatic Rebase
      uses: cirrus-actions/rebase@1.3.1
```

**Comment by timbuckley** [comment]
@awasum Unfortunately the issue is this job requires a `GITHUB_TOKEN` environment variable to be set in Settings > Secrets for this repo. But I believe the only person who has permission to add any env vars to Secrets is @ianstormtaylor

**Comment by awasum** [comment]
> @awasum Unfortunately the issue is this job requires a `GITHUB_TOKEN` environment variable to be set in Settings > Secrets for this repo. But I believe the only person who has permission to add any env vars to Secrets is @ianstormtaylor

@timbuckley . Github Automatically creates a the token for us when we enable Github Actions on our repos. We dont need to do anything else. See : https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token

File: .github/workflows/rebase.yml (hunk 1 above)
Comment (awasum) [reply .github/workflows/rebase.yml:18]: Ohh..So it will still work like that? I did not know that. I will commit your suggestions @timbuckley

**Pull request merged**